doc/workshops/scientific-computing.rst

```diff
@@ -2,4 +2,698 @@
+
+A hypothesis is an statement that can be tested by means of the scientific method. They must be verifiable using empirical evidence or data. For example, *green is the best color* is not a hypothesis since it cannot be proven or disproven.
+
+With respect to the above theory of a person's religiosity, to explore it we should find two variables to relate. The first variable would be related to a **person's religious feeling**. The second variable would be related to a person's **ability to solve problems and situations**.
+
+The OKCupid dataset has useful variables that we can use to investigate this theory. For example, question q41 (*How important is religion/God in your life?*) provides information about a person's religious sentiment. The second variable is a slightly more difficult to come up with, as it is somewhat more complex. Fortunately, the researchers who collected the dataset have also selected 14 questions that can serve as a proxy for a cognitive ability test. This selection of questions can be found in the file "test_items.csv". [ADD LINK TO DOWNLOAD test_items.csv]
+
+.. code-block: python
+   
+   test_items = pd.read_csv("test_items.csv")
+
+So, let's try to see if there is any evidence in the OKCupid dataset to support the following hypothesis:
+
+*Nonreligious groups (atheists/agnostics) would score higher on tests of cognitive ability*.
+
+
+
+==============
+Build the test
+==============
+
+We already know which questions will be part of the cognitive ability test. Now we are going to process that data to obtain the scores for each person.
+
+Make a copy of the original dataset:
+
+.. code-block: python
+
+   ca_test = data.copy()
+
+The next step is to find the correct answer to each test question (these answers are in ``test_items``).
+
+.. code-block: python
+
+   right_answers = []
+   for i, j in test_items.iterrows():
+       right_answers.append(j.iloc[j["option_correct"] + 2])
+   test_items["right_answer"] = right_answers
+
+The correct answers have been stored in the test_items dataframe.
+
+Next, we will indicate, in the ca_test dataframe, whether the person answered correctly to each question selected in the cognitive ability test. These answers will be entered as a boolean in new variables with the prefix "resp_" followed by the corresponding question code. The Boolean is calculated with the function ``lambda row: row[q] == a, axis=1``.
+
+.. code-block: python
+
+   for i, j in test_items.iterrows():
+       q = "q" + str(j["ID"])
+       a = str(j["right_answer"])
+       try:
+           ca_test.dropna(subset=[q], inplace=True)
+           ca_test["resp_" + q] = ca_test.apply(lambda row: row[q] == a, axis=1)
+       except:
+           print(f"{q} not found.")
+           pass
+
+.. note::
+
+   Some test_items questions are not in the data, but don't worry about that. We use ``try... except`` to ignore these errors. 
```

Comment:
```suggestion
   Some ``test_items`` questions are not in the data, but don't worry about that. We use a ``try... except`` block to ignore these errors. 
```

I understand, but I think we may need a quick justification for this. Otherwise the user may be left wondering if they misunderstood something.

Reply:
Ah! I think this is explained below. I'm wondering then why we have the above sentence - it may be better to include this observation in the "important" block below.